src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -79,9 +88,39 @@ export default class Pools {
     } catch {
       // eslint-disable-previous-line no-empty
     }
+
+    if (this.excludedAddresses == null) {
+      // fetch only once
+      this.excludedAddresses = await this.getExcludedAddresses();
+    }
+
     return this.serialize(pools, pastPools, period, prices, currency);
   }
 
+  public removeExcludedAddressesFromTotalLiquidty(
+    pool: Pool,
+    totalLiquidity: string
+  ) {
+    let _totalLiquidity = bnum(totalLiquidity);
```

Comment:
This is a handy function to apply the removal of addresses externally (when having a custom `totalLiquidity` like stable pools).

For simplicity - it assumes the decoration phase already ran and so `this.excludedAddresses` is defined.